src/ast.rs

```diff
@@ -81,6 +81,27 @@ impl Value {
             Value::Regex(_) => Type::Regex,
         }
     }
+
+    pub fn as_string(&self) -> Option<&String> {
```

Comment:
Rust always suggests `&str` instead of `&String`.

Reply:
Rust often suggests using &str (a string slice) instead of &String (a reference to a String) in function signatures and other places where you're borrowing string data

&str is more general: A &str can point to either a String (which owns its data on the heap) or a string literal (which is typically stored in the program's binary).

&String is more specific: A &String can only refer to a String.


By accepting &str, your function becomes more versatile. It can work with both types of string data without requiring the caller to explicitly convert a string literal to a String.

Reply:
More versatile for your code, if you use `str`